code/controllers/subsystems/processing/overmap.dm

```diff
@@ -0,0 +1,25 @@
+// Some duplicated behavior from SSprocessing in here so atoms can 
+// still process normally as well as handling overmap movement.
+
+SUBSYSTEM_DEF(overmap)
+	name = "Overmap"
+	priority = SS_PRIORITY_OVERMAP
+	wait = 2
+	flags = SS_BACKGROUND | SS_POST_FIRE_TIMING | SS_NO_INIT
+	var/list/moving_entities = list()
+	var/list/current_run = list()
+
+// Processing boilerplate.
+/datum/controller/subsystem/overmap/fire(resumed = 0)
+	if(!resumed)
+		src.current_run = moving_entities.Copy()
+	var/list/current_run = src.current_run
+	var/wait = src.wait
+	var/times_fired = src.times_fired
+	while(current_run.len)
+		var/obj/effect/overmap/entity = current_run[current_run.len]
+		current_run.len--
+		if(QDELETED(entity) || entity.ProcessOvermap(wait, times_fired) == PROCESS_KILL)
```

Comment:
Does `ProcessOvermap` have sanity checks for being called on a stationary object?

Reply:
It doesn't need them, a stationary object will return PROCESS_KILL or simply not move.

Reply:
````
/obj/effect/overmap/proc/ProcessOvermap(wait, tick)

	if(halted || is_still())
		return PROCESS_KILL

	if(!can_move)
		return
````